### ISSUE TITLE
NAS Volume1 unavailable to select files to backup in CP interface

### ISSUE THREAD

**Issue by jvarrone** [jlesage/docker-crashplan-pro#230]
Still having trouble getting docker/cp reloaded. I figured out the problems getting the ports set correct for the GUI. But I couldn't sync to the CP Repository. With a conversation with CP tech, he said the server can't locate the previously selected files on the NAS (all files in the CP console at CP are showing as deleted). So I started from scratch. Removed the container, images, cache, etc. Reloaded Docker and the jl-crashplanpro-latest. When I went to manage files I can no longer pick from the Synology "Volume1" location. All I see is "root" and directories below it. Even with show hidden files checked. Clearly this is why I couldn't sync up. I tried creating a new container with elevated permissions, the same problem. See below screenshot of what the CP interface sees - no Volume1 to pick from! I'm really in a bad way since I've had no backups for weeks now.

![Capture](https://user-images.githubusercontent.com/3275462/77367050-b83b3580-6d2f-11ea-9131-e1729792d28f.JPG)

**Comment by jvarrone** [comment]
Uninstalled, re-installed, still cannot see "volume1" to either manage files for backup or restore to. I can show hidden files at CP Central, pick a file to restore to "original location" but it goes nowhere. If I select "other location" to restore to then "volume1" still does not show up.

I did a complete "find" on the NAS and the file did not get restored. What I did to test this is I deleted an outdated, no longer needed file, on the NAS and then restored it from CP Central using the CP interface on the NAS. File not found on the NAS.

I realize the docker and CP interface is a free product, which I appreciate, but CP tech will provide no help because they claim it unsupported. I am willing to pay to have this straightened out. I'm really in a bad way - I cannot back up from - or restore to the NAS.

**This is what I get if I choose "other location"**
![cp1](https://user-images.githubusercontent.com/3275462/77429180-a4cfaf00-6daf-11ea-8627-faa64784c4be.JPG)

**This is what I get if I try to access root:**

![cp2](https://user-images.githubusercontent.com/3275462/77429240-b9ac4280-6daf-11ea-8962-6507eddf7739.JPG)

**Comment by jvarrone** [comment]
Somehow I got it to run and see volume1. But the volume's location doesn't match that on Cp Central.
Cp Central shows volume one as /volume1
But docker/CP shows it as /root/volume1 and next to all the directories on the far right the CP interface shows "file not found" which is likely why CP Central can't see the directories.

**Please help - I'm desparate**

**Comment by jlesage** [comment]
You can map a folder from your Synology to a folder in the container.  This is how you expose your data to the container.  I think that Synology calls this "Shared folders".
So if you mapped `volume1` to `/root/volume1` in the container, CP will see your data in `/root/volume1`.  If you want CP to see your files in `/volume1`, you need to adjust your mapping...

**Comment by jvarrone** [comment]
I've read the docs from head to bottom. No mention of how to do this in the GUI. So I deleted the container, re-created it and went to advanced settings, volume tab. All my directories below /volume1 are showing there just fine. Nothing there to map and I never mapped /volume1 to /root/volume1 previously on my own. Once I finish creating the container is there any config file where I can check what, if anything, was remapped?
Thank you for continuing to help. I feel like we are so close to getting this fixed...

**Comment by jlesage** [comment]
Ok so when you edit the container, you should have a tab `Volume`.  This is where you can map a folder to the container.  The `Mount path` is the path inside the container, i.e. the path seen by CrashPlan.

Example of this page:

https://preview.redd.it/t8xaba2mdp111.png?width=646&format=png&auto=webp&s=2b4e191ed3b4cace958440a53d95bce7706f05bd

**Comment by jvarrone** [comment]
sorry to be a pest, but based on my paths what do I put on the left and right side of the volume mapping?
to reiterate:
CP Central shows /volume1
CP Client shows /root/volume1

I've been using docker/crashplan for quite a while. I never had issues with it until now. I was always able to upgrade without a problem. I can only surmise CP messed something up with their latest version.

**Comment by jlesage** [comment]
I guess you need to click "Add folder", select the `volume1` folder and then set the mount point to `/volume1`.

**Comment by jvarrone** [comment]
Yes, I tried that but the odd thing is that the folders on the left side to pick from are those already below /volume1 (image attached). I'm totally lost. I am willing to pay to have this fixed.

![Capture](https://user-images.githubusercontent.com/3275462/77680747-c32dda00-6f6a-11ea-9725-8c5672eef6d4.JPG)

**Comment by jlesage** [comment]
So probably that you created the container via shell the first time ?

Do you need to backup every available folders?

**Comment by jvarrone** [comment]
No, I created the container always via the GUI. I never had any trouble before now updating to a new version.

I do not backup every available folder on the NAS.

**Comment by jlesage** [comment]
Are you expecting your Synology folders to be under volume1 in CrashPlan ?  For example, are you expecting `Photos` to be found at `/volume1/Photos` in CP ?

If yes, then just add all the volume you need, making sur to set the `Mount path` to `/volume1/XXX`.  For example, your `Photos` folder should have a mount point set to `/volume1/Photos`.

**Comment by jvarrone** [comment]
Do I have to do this for every folder I want to backup? Or can I just set the mount point to /volume1 so I can pick files to backup from the GUI? Also, what is the relation of this to the /config default in the container? In all the time I've been using the docker and crashplan image I've never had to do anything like this. I'm totally confused.

**Comment by jlesage** [comment]
> Do I have to do this for every folder I want to backup? Or can I just set the mount point to /volume1 so I can pick files to backup from the GUI? 

Yes this is needed for each folder, because you cannot have multiple volumes with the same mount point.

> Also, what is the relation of this to the /config default in the container? 

`/config` is the directory inside the container where all states of CrashPlan are saved.  When mapped to a directory on your Synology, it allows you to update or re-create the container without having to start over with CrashPlan.  You have to know that anything written by an application inside a container is not permanent, unless data is written to a directory mapped to the host.

> In all the time I've been using the docker and crashplan image I've never had to do anything like this. I'm totally confused.

Sorry, hard to comment on this since I don't know how you were setup before :)  But mapping volumes to a container is the only way to either share your data with the container or to save persistent data needed by the container.

**Comment by KipiKland** [comment]
If we refer to y Synology NAS i have done a new config last week that works perfectly.
Created a Crashplan user "Crashplan" and a "Backup operator" group. **Granted backup operator group RO access to all shares part of volume 1.** Gathered user id and group ID for backup operator group.
Then confogured the container with SSH:

docker create --name=crashplan-pro \
    -p 5800:5800 \
    -v /volume1/docker/crashplan:/config:rw \
    -v /volume1/:/volume1:ro \
    -e USER_ID=1031 \
    -e GROUP_ID=65548 \
    -e CLEAN_TMP_DIR=1 \
    -e CRASHPLAN_SRV_MAX_MEM=4096M \
    -e TZ=Europe/Zurich \
    jlesage/crashplan-pro

-Make sure User ID is properly set as well as group ID which should be the group ID of backup operators (since user may be member of many groups)
- You will NOT see the volume configuration association between Volume 1 and Volume 1 on Syno NAS. However this volume association WILL work. 
Problem solved i have access to root, and volume 1 is visible.

**Comment by jvarrone** [comment]
Hi,
How do I definitively determine numbers for:
-e USER_ID=1031
-e GROUP_ID=65548 

for my particular setup?

**Comment by jvarrone** [comment]
also, what is the timezone text for USA Eastern New York?

**Comment by KipiKland** [comment]
To find a user's UID or GID in Unix, use the id command. To find a specific user's UID, at the Unix prompt, enter: 
 id -u username

Replace username with the appropriate user's username. To find a user's GID, at the Unix prompt, enter:
 id -g username

If you wish to find out all the groups a user belongs to, instead enter:

 id -G username
If you wish to see the UID and all groups associated with a user, enter id without any options, as follows:

 id username

For the time zone, your is America/New_York
please refer to the following webpage: [https://en.wikipedia.org/wiki/List_of_tz_database_time_zones](url)

**Comment by jvarrone** [comment]
I will save the info on timezones. Very helpful.
What username am I looking for? Is it the username I used to telnet into the NAS to perform the docker create commands?

Thanks!

**Comment by KipiKland** [comment]
> Nevermind - I just answered my own question. Thank you again for your help!
> Be safe...

No worries :) 
It is better to associate the container with a known user on the NAS. When manipulating files, the container will do it in the context of the user in question, hence having its rights and privileges. 
Usually you dont grant a user rights, you grant a group rights where a user is member of, this is the clean way. 
have a look at this: [](https://medium.com/@mccode/understanding-how-uid-and-gid-work-in-docker-containers-c37a01d01cf)

**Comment by jvarrone** [comment]
Kip - is it possible to PM here or have you contact me directly?
Thanks - Jim

**Comment by KipiKland** [comment]
> Kip - is it possible to PM here or have you contact me directly?
> Thanks - Jim
What is exactly your issue ?

**Comment by jvarrone** [comment]
I'm having one problem after another. I've been using Docker/lasage for about two years now and never had this much trouble getting it setup. It all started back about 45 days ago when I needed to perform an update. I followed the same steps I always had but it didn't work this time. Since then I've deleted and reinstalled Docker several times and I still can't get it to work. I even used CL interface to manually remove containers. At this point I need some professional help. I am even willing to pay to get it running again. CP support is zero help because of the Docker environment.
I'm really stuck in cement at this point.

**Comment by KipiKland** [comment]
i am an IT pro but not a linux pro, however i can try. First let me know what is exactly not working today, can you provide precise details so i can make initial diag.
